### ISSUE TITLE
withFields() method causes invalid URL

### ISSUE THREAD

**Issue by remy-phelipot** [fabric8io/kubernetes-client#381]
Hello,

I encoutered a bug with the withFields() method. If multiple elements are added in the map used by this method, the generated URL is malformed. 

Here is a minimal example of this issue:
```java
import io.fabric8.kubernetes.api.model.Pod;
import io.fabric8.kubernetes.client.*;

import java.util.HashMap;
import java.util.Map;


public class Test {
    public static void main(String[] args) {
        KubernetesClient client = new DefaultKubernetesClient();
        Map<String,String> selector = new HashMap<String,String>();
        selector.put("metadata.namespace","default");
        selector.put("metadata.name","c1internals29iz-nrgcb");

        Watch watch = client.pods()
                .inNamespace("default")
                .withFields(selector)
                .watch(new Watcher<Pod>() {
            public void eventReceived(Action action, Pod pod) {
                System.out.println(pod.getMetadata().getResourceVersion());
            }

            public void onClose(KubernetesClientException e) {
                System.out.println("client closed");
            }
        });

        watch.close();
        client.close();
    }
}
```

An exception is thrown during the execution:
```
Exception in thread "main" io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: GET at: http://localhost:44000/api/v1/namespaces/default/pods?fieldSelector=metadata.name%3Dc1internals29iz-nrgcbmetadata.namespace%3Ddefault. Message: invalid selector: 'metadata.name=c1internals29iz-nrgcbmetadata.namespace=default'; can't understand 'metadata.name=c1internals29iz-nrgcbmetadata.namespace=default'. Received status: Status(apiVersion=v1, code=500, details=null, kind=Status, message=invalid selector: 'metadata.name=c1internals29iz-nrgcbmetadata.namespace=default'; can't understand 'metadata.name=c1internals29iz-nrgcbmetadata.namespace=default', metadata=ListMeta(resourceVersion=null, selfLink=null, additionalProperties={}), reason=null, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:290)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:243)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:212)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:383)
	at io.fabric8.kubernetes.client.dsl.internal.WatchConnectionManager.<init>(WatchConnectionManager.java:67)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch(BaseOperation.java:476)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch(BaseOperation.java:471)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.watch(BaseOperation.java:57)
	at Test.main(Test.java:20)
```

There is no comma between the parameter elements.

**Comment by remy-phelipot** [comment]
My bad, it has been already solved in #372.

**Comment by jimmidyson** [comment]
Thanks for the report though!
